gnovm/stdlibs/std/coins.gno

```diff
@@ -13,21 +15,77 @@ type Coin struct {
 	Amount int64  `json:"amount"`
 }
 
+// NewCoin returns a new coin with a denomination and amount
+func NewCoin(denom string, amount int64) Coin {
```

Comment:
This is the ordering in the whole codebase, I believe that if we choose to switch these we should do it in a separate PR.

Reply:
I feel it will be a problem soon, but we can leave it as it is for historical reasons...

Reply:
@ajnavarro +1

Reply:
@leohhhn 
How are they reversed, there was never a constructor before? We always initialized objects directly (since these are public fields)

I think it's fine to swap them in this PR

Reply:
As stated in the PR description, this functionality is ported/adapted from `coins.go` in TM2. This is the way they are over there. Sure, I can swap them if needed, but I see no issue with first specifying the denom and then the amount.

Reply:
Why not swap it for all occurrences?

Reply:
@zivkovicmilos Out of scope